project_template/src/config/settings.js

```diff
@@ -1 +1,4 @@
-export default require(`./settings.${process.env.NODE_ENV}.json`);
+const envSpecificSettings = require(`./settings.${process.env.NODE_ENV}.json`);
+const defaults = require("./settings.default.json");
+
+export default Object.assign(defaults, envSpecificSettings);
```

Comment:
Object.assign is modifying `defaults` here. I'm not entirely sure but my guess is it would actually globally modify the value of `require("./settings.default.json")`) since export statements are only evaluated once.

So to be on the safe side I think it would be better to make this `Object.assign({}, defaults, envSpecificSettings)`

Reply:
You're right, this has an unintended side effect. I didn't realize that require("./settings.default.json") in one file returns the same object as require("./settings.default.json") in another file. I'll commit an updated version of this file.